client/lib/cart/store/cart-synchronizer.js

```diff
@@ -106,17 +107,27 @@ CartSynchronizer.prototype.update = function( changeFunction ) {
 		// with a consistent state when we apply our local changes. The strategy
 		// here is to queue the changes until the first request has completed, then
 		// proceed as normal.
-		this._enqueueChange( changeFunction );
+		this._enqueueChange( changeFunction, serverFlushCallback );
+		if ( ! this._activeRequest ) {
+			this.fetch();
+		}
+		debug( 'update: change enqueued' );
 		return;
 	}
 
+	debug( 'update: processing change' );
+	this._addFlushCallback( serverFlushCallback );
+	this._processChangeFunction( changeFunction );
+	this.emit( 'change' );
+};
+
+CartSynchronizer.prototype._processChangeFunction = function ( changeFunction ) {
 	if ( this._activeRequest && this._activeRequest.state !== 'canceled' ) {
 		this._activeRequest.state = 'canceled';
 	}
 
 	this._latestValue = changeFunction( this._latestValue );
 	this._performRequest( 'update', this._postToServer.bind( this ) );
-	this.emit( 'change' );
```

Comment:
I guess it's fine as it is, my idea was to emit `synced` event in the CartSynchronizer that the CartStore could listen to.
So replace this call:
```js
_synchronizer.update( wrappedFunction, serverFlushCallback );
```
by:
```js
_synchronizer.update( wrappedFunction );
_synchronizer.once( 'synced', serverFlushCallback );
```